src/Medidata.ZipkinTracer.Core/ZipkinClient.cs

```diff
@@ -69,15 +67,15 @@ public Span StartClientTrace(Uri remoteUri, string methodName)
                 catch (Exception ex)
                 {
                     logger.Error("Error Starting Client Trace", ex);
+                return null;
                 }
             }
-            return null;
-        }
 
         public void EndClientTrace(Span clientSpan, int statusCode)
         {
-            if (isTraceOn)
-            {
+            if (!isTraceOn)
+                return;
+
                 try
```

Comment:
Thanks, I have to practice my conflict resolution skills :sweat_smile:. Fixed now.